app/src/main/java/com/example/keniphhomemaintenance/dwellings/DwellingsScreenView.kt

```diff
@@ -0,0 +1,226 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.KeyboardArrowDown
+import androidx.compose.material.icons.filled.KeyboardArrowUp
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.toSize
+import androidx.compose.ui.window.Dialog
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.NavController
+
+
+val addDwellingDialog = mutableStateOf(false)
+
+@ExperimentalComposeUiApi
+@Composable
+fun DwellingsScreenView(navController: NavController) {
+
+    ConstraintLayout {
+
+        val (lazyColumn, scaffold) = createRefs()
+
+        val listOfDwellings = dwellingsViewModel.dwellingsList.value
+
+        Scaffold(
+
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .background(Color.Transparent)
+                .constrainAs(scaffold) {
+                    bottom.linkTo(parent.bottom, margin = 16.dp)
+                },
+            backgroundColor = Color.Transparent,
+            floatingActionButton = {
+                FloatingActionButton(
+                    modifier = Modifier.size(50.dp),
+                    shape = CircleShape,
+                    onClick = {
+                        addDwellingDialog.value = true
+                        // dwellingsViewModel.addDwelling()
+                    }) {
+                    Icon(Icons.Default.Add, contentDescription = "Button to add a dwelling")
+                }
+            },
+            floatingActionButtonPosition = FabPosition.Center
+        ) {
+            if (addDwellingDialog.value) {
+                NewDwellingDialog()
+            }
+        }
+        LazyColumn(
+            modifier = Modifier
+                .constrainAs(lazyColumn) {
+                    top.linkTo(parent.top)
+                }
+                .padding(bottom = 100.dp),
+
+            contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp)
+        ) {
+            items(
+                items = listOfDwellings,
+                itemContent = {
+                    DwellingItem(dwelling = it, navController)
+                }
+            )
+        }
+
+    }
+}
+
```

Comment:
Below is the dialog box that appears over this screen. Unsure if it should be in its own file

Reply:
I'd put it in a different file, yah